dsc/docs-conceptual/dsc-2.0/tutorials/create-dsc-resource-machine-config.md

```diff
@@ -0,0 +1,2086 @@
+```
+
+```Output
+    Directory: C:\code\dsc\ExampleResources
+
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+-a---            9/8/2022  1:58 PM              0 Helpers.ps1
+```
+
+Open `Helpers.ps1` in VS Code. Add the following line, replacing `<Separator>` with `;` if you're on
+Windows and `:` if you're on Linux or macOS.
+
+```powershell
+$env:PSModulePath += "<Separator>$pwd"
+```
+
+Open `ExampleResources.psm1` in VS Code. The module is now scaffolded and ready for you to author a
+DSC Resource.
+
+## 2 - Add a class-based DSC Resource
+
+To define a class-based DSC Resource, we write a PowerShell class in a module file and add the
+**DscResource** attribute to it.
+
+### Define the class
+
+In `ExampleResources.psm1`, add the following code:
+
+```powershell
+[DscResource()]
+class Tailspin {
+
+}
+```
+
+This code adds `Tailspin` as a class-based DSC Resource to the **ExampleResources** module.
+
+Hover on `[DscResource()]` and read the warnings.
+
+:::image type="complex" source="media/create-class-based-resource/Empty-Class-Warnings.png" alt-text="Screenshot of the DSCResource attribute's warnings in VS Code.":::
+Hovering on the DSCResource attribute displays four warnings. 1. The DSC Resource 'Tailspin' is missing a `Set()` method that returns `[void]` and accepts no parameters. 2. The DSC Resource 'Tailspin' is missing a `Get()` method that returns `[Tailspin]` and accepts no parameters. 3. The DSC Resource 'Tailspin' is missing a `Test()` method that returns `[bool]` and accepts no parameters. 4. The DSC Resource 'Tailspin' must have at least one key property (using the syntax `[DscProperty(Key)]`.)
+:::image-end:::
+
+These warnings list the requirements for the class to be valid DSC Resource.
+
+### Minimally implement required methods
+
+Add a minimal implementation of the `Get()`, `Test()`, and `Set()` methods to the class.
+
+```powershell
+class Tailspin {
+    [Tailspin] Get() {
+        $CurrentState = [Tailspin]::new()
+        return $CurrentState
+    }
+
+    [bool] Test() {
+        return $true
+    }
+
+    [void] Set() {}
+}
+```
+
+With the methods added, the **DscResource** attribute only warns about the class not having a
+**Key** property.
+
+## 3 - Define DSC Resource properties
+
+You should define the properties of the DSC Resource before the methods. The properties define the
+manageable settings for the DSC Resource. They're used in the methods.
+
+### Understand the TSToy application
+
+Before you can define the properties of your DSC Resource, you need to understand what settings you
+want to manage.
+
+For this tutorial, we're defining a DSC Resource for managing the settings of the fictional TSToy
+application through its configuration file. TSToy is an application that has configuration at the
+user and machine levels. The DSC Resource should be able to configure either file.
+
+The DSC Resource should enable users to configure:
+
```

Comment:
I wonder if a novice might look at this list and think they are "DSC" concepts rather than properties of the fictional app?

I don't have great suggestions though. DB connection strings or maybe what port the app will listen on. We've never had a go-to fictional app, unfortunately.

Reply:
Maybe something like:

> The DSC Resource should enable users to define the following settings for TailSpin: